### DISCUSSION TITLE
AX6 编译出错,求解

### DISCUSSION THREAD

**Discussion by chy001a** [Boos4721/openwrt#108] [Q&A/答疑解惑] [answered]
make[3]: Entering directory '/home/chy/openwrt/feeds/routing/batctl'
touch /home/chy/openwrt/build_dir/target-aarch64_cortex-a53_musl/batctl-default/batctl-2022.0/.prepared_233e21746e8a5649b56275f2f14d8549_6664517399ebbbc92a37c5bb081b5c53_check
. /home/chy/openwrt/include/shell.sh; gzip -dc /home/chy/openwrt/dl/batctl-2022.0.tar.gz | tar -C /home/chy/openwrt/build_dir/target-aarch64_cortex-a53_musl/batctl-default/batctl-2022.0/.. -xf -
[ ! -d ./src/ ] || cp -fpR ./src/. /home/chy/openwrt/build_dir/target-aarch64_cortex-a53_musl/batctl-default/batctl-2022.0
touch /home/chy/openwrt/build_dir/target-aarch64_cortex-a53_musl/batctl-default/batctl-2022.0/.prepared_233e21746e8a5649b56275f2f14d8549_6664517399ebbbc92a37c5bb081b5c53
rm -f /home/chy/openwrt/build_dir/target-aarch64_cortex-a53_musl/batctl-default/batctl-2022.0/.configured_*
rm -f /home/chy/openwrt/staging_dir/target-aarch64_cortex-a53_musl/stamp/.batctl.default_installed
(cd /home/chy/openwrt/build_dir/target-aarch64_cortex-a53_musl/batctl-default/batctl-2022.0/./; if [ -x ./configure ]; then find /home/chy/openwrt/build_dir/target-aarch64_cortex-a53_musl/batctl-default/batctl-2022.0/ -name config.guess | xargs -r chmod u+w; find /home/chy/openwrt/build_dir/target-aarch64_cortex-a53_musl/batctl-default/batctl-2022.0/ -name config.guess | xargs -r -n1 cp --remove-destination /home/chy/openwrt/scripts/config.guess; find /home/chy/openwrt/build_dir/target-aarch64_cortex-a53_musl/batctl-default/batctl-2022.0/ -name config.sub | xargs -r chmod u+w; find /home/chy/openwrt/build_dir/target-aarch64_cortex-a53_musl/batctl-default/batctl-2022.0/ -name config.sub | xargs -r -n1 cp --remove-destination /home/chy/openwrt/scripts/config.sub; AR="aarch64-openwrt-linux-musl-gcc-ar" AS="ccache_cc -c -Os -pipe -mcpu=cortex-a53 -fno-caller-saves -fno-plt -fhonour-copts -Wno-error=unused-but-set-variable -Wno-error=unused-result -fmacro-prefix-map=/home/chy/openwrt/build_dir/target-aarch64_cortex-a53_musl/batctl-default/batctl-2022.0=batctl-2022.0 -Wformat -Werror=format-security -fstack-protector -D_FORTIFY_SOURCE=1 -Wl,-z,now -Wl,-z,relro -ffunction-sections -fdata-sections -flto" LD=aarch64-openwrt-linux-musl-ld NM="aarch64-openwrt-linux-musl-gcc-nm" CC="ccache_cc" GCC="ccache_cc" CXX="ccache_cxx" RANLIB="aarch64-openwrt-linux-musl-gcc-ranlib" STRIP=aarch64-openwrt-linux-musl-strip OBJCOPY=aarch64-openwrt-linux-musl-objcopy OBJDUMP=aarch64-openwrt-linux-musl-objdump SIZE=aarch64-openwrt-linux-musl-size CFLAGS="-Os -pipe -mcpu=cortex-a53 -fno-caller-saves -fno-plt -fhonour-copts -Wno-error=unused-but-set-variable -Wno-error=unused-result -fmacro-prefix-map=/home/chy/openwrt/build_dir/target-aarch64_cortex-a53_musl/batctl-default/batctl-2022.0=batctl-2022.0 -Wformat -Werror=format-security -fstack-protector -D_FORTIFY_SOURCE=1 -Wl,-z,now -Wl,-z,relro -ffunction-sections -fdata-sections -flto " CXXFLAGS="-Os -pipe -mcpu=cortex-a53 -fno-caller-saves -fno-plt -fhonour-copts -Wno-error=unused-but-set-variable -Wno-error=unused-result -fmacro-prefix-map=/home/chy/openwrt/build_dir/target-aarch64_cortex-a53_musl/batctl-default/batctl-2022.0=batctl-2022.0 -Wformat -Werror=format-security -fstack-protector -D_FORTIFY_SOURCE=1 -Wl,-z,now -Wl,-z,relro -ffunction-sections -fdata-sections -flto " CPPFLAGS="-I/home/chy/openwrt/staging_dir/toolchain-aarch64_cortex-a53_gcc-11.2.0_musl/usr/include -I/home/chy/openwrt/staging_dir/toolchain-aarch64_cortex-a53_gcc-11.2.0_musl/include/fortify -I/home/chy/openwrt/staging_dir/toolchain-aarch64_cortex-a53_gcc-11.2.0_musl/include " LDFLAGS="-L/home/chy/openwrt/staging_dir/toolchain-aarch64_cortex-a53_gcc-11.2.0_musl/usr/lib -L/home/chy/openwrt/staging_dir/toolchain-aarch64_cortex-a53_gcc-11.2.0_musl/lib -znow -zrelro -Wl,--gc-sections -fuse-linker-plugin "   ./configure --target=aarch64-openwrt-linux --host=aarch64-openwrt-linux --build=x86_64-pc-linux-gnu --program-prefix="" --program-suffix="" --prefix=/usr --exec-prefix=/usr --bindir=/usr/bin --sbindir=/usr/sbin --libexecdir=/usr/lib --sysconfdir=/etc --datadir=/usr/share --localstatedir=/var --mandir=/usr/man --infodir=/usr/info --disable-nls  ; fi; )
touch /home/chy/openwrt/build_dir/target-aarch64_cortex-a53_musl/batctl-default/batctl-2022.0/.configured_68b329da9893e34099c7d8ad5cb9c940
rm -f /home/chy/openwrt/build_dir/target-aarch64_cortex-a53_musl/batctl-default/batctl-2022.0/.built
touch /home/chy/openwrt/build_dir/target-aarch64_cortex-a53_musl/batctl-default/batctl-2022.0/.built_check
CFLAGS="-Os -pipe -mcpu=cortex-a53 -fno-caller-saves -fno-plt -fhonour-copts -Wno-error=unused-but-set-variable -Wno-error=unused-result -fmacro-prefix-map=/home/chy/openwrt/build_dir/target-aarch64_cortex-a53_musl/batctl-default/batctl-2022.0=batctl-2022.0 -Wformat -Werror=format-security -fstack-protector -D_FORTIFY_SOURCE=1 -Wl,-z,now -Wl,-z,relro -ffunction-sections -fdata-sections -flto  -I/home/chy/openwrt/staging_dir/toolchain-aarch64_cortex-a53_gcc-11.2.0_musl/usr/include -I/home/chy/openwrt/staging_dir/toolchain-aarch64_cortex-a53_gcc-11.2.0_musl/include/fortify -I/home/chy/openwrt/staging_dir/toolchain-aarch64_cortex-a53_gcc-11.2.0_musl/include " CXXFLAGS="-Os -pipe -mcpu=cortex-a53 -fno-caller-saves -fno-plt -fhonour-copts -Wno-error=unused-but-set-variable -Wno-error=unused-result -fmacro-prefix-map=/home/chy/openwrt/build_dir/target-aarch64_cortex-a53_musl/batctl-default/batctl-2022.0=batctl-2022.0 -Wformat -Werror=format-security -fstack-protector -D_FORTIFY_SOURCE=1 -Wl,-z,now -Wl,-z,relro -ffunction-sections -fdata-sections -flto  -I/home/chy/openwrt/staging_dir/toolchain-aarch64_cortex-a53_gcc-11.2.0_musl/usr/include -I/home/chy/openwrt/staging_dir/toolchain-aarch64_cortex-a53_gcc-11.2.0_musl/include/fortify -I/home/chy/openwrt/staging_dir/toolchain-aarch64_cortex-a53_gcc-11.2.0_musl/include " LDFLAGS="-L/home/chy/openwrt/staging_dir/toolchain-aarch64_cortex-a53_gcc-11.2.0_musl/usr/lib -L/home/chy/openwrt/staging_dir/toolchain-aarch64_cortex-a53_gcc-11.2.0_musl/lib -znow -zrelro -Wl,--gc-sections -fuse-linker-plugin " LIBNL_NAME="libnl-tiny" LIBNL_GENL_NAME="libnl-tiny" make  -C /home/chy/openwrt/build_dir/target-aarch64_cortex-a53_musl/batctl-default/batctl-2022.0/. AR="aarch64-openwrt-linux-musl-gcc-ar" AS="ccache_cc -c -Os -pipe -mcpu=cortex-a53 -fno-caller-saves -fno-plt -fhonour-copts -Wno-error=unused-but-set-variable -Wno-error=unused-result -fmacro-prefix-map=/home/chy/openwrt/build_dir/target-aarch64_cortex-a53_musl/batctl-default/batctl-2022.0=batctl-2022.0 -Wformat -Werror=format-security -fstack-protector -D_FORTIFY_SOURCE=1 -Wl,-z,now -Wl,-z,relro -ffunction-sections -fdata-sections -flto" LD=aarch64-openwrt-linux-musl-ld NM="aarch64-openwrt-linux-musl-gcc-nm" CC="ccache_cc" GCC="ccache_cc" CXX="ccache_cxx" RANLIB="aarch64-openwrt-linux-musl-gcc-ranlib" STRIP=aarch64-openwrt-linux-musl-strip OBJCOPY=aarch64-openwrt-linux-musl-objcopy OBJDUMP=aarch64-openwrt-linux-musl-objdump SIZE=aarch64-openwrt-linux-musl-size CROSS="aarch64-openwrt-linux-musl-" ARCH="aarch64" REVISION="2022.0-openwrt-1" $V_batctl_config ;
make[4]: Entering directory '/home/chy/openwrt/build_dir/target-aarch64_cortex-a53_musl/batctl-default/batctl-2022.0'
ccache_cc -Os -pipe -mcpu=cortex-a53 -fno-caller-saves -fno-plt -fhonour-copts -Wno-error=unused-but-set-variable -Wno-error=unused-result -fmacro-prefix-map=/home/chy/openwrt/build_dir/target-aarch64_cortex-a53_musl/batctl-default/batctl-2022.0=batctl-2022.0 -Wformat -Werror=format-security -fstack-protector -D_FORTIFY_SOURCE=1 -Wl,-z,now -Wl,-z,relro -ffunction-sections -fdata-sections -flto  -I/home/chy/openwrt/staging_dir/toolchain-aarch64_cortex-a53_gcc-11.2.0_musl/usr/include -I/home/chy/openwrt/staging_dir/toolchain-aarch64_cortex-a53_gcc-11.2.0_musl/include/fortify -I/home/chy/openwrt/staging_dir/toolchain-aarch64_cortex-a53_gcc-11.2.0_musl/include  -Wall -W -std=gnu99 -fno-strict-aliasing -MD -MP -I/home/chy/openwrt/staging_dir/target-aarch64_cortex-a53_musl/usr/include/libnl-tiny  -I/home/chy/openwrt/staging_dir/target-aarch64_cortex-a53_musl/usr/include/libnl-tiny  -D_GNU_SOURCE -DSOURCE_VERSION=\"2022.0-openwrt-1\"  -c -o bat-hosts.o bat-hosts.c
ccache: error: Failed to read from file stream
make[4]: *** [Makefile:162: bat-hosts.o] Error 1
make[4]: Leaving directory '/home/chy/openwrt/build_dir/target-aarch64_cortex-a53_musl/batctl-default/batctl-2022.0'
make[3]: *** [Makefile:262: /home/chy/openwrt/build_dir/target-aarch64_cortex-a53_musl/batctl-default/batctl-2022.0/.built] Error 2
make[3]: Leaving directory '/home/chy/openwrt/feeds/routing/batctl'
time: package/feeds/routing/batctl/default/compile#0.34#0.05#0.34
    ERROR: package/feeds/routing/batctl failed to build (build variant: default).
make[2]: *** [package/Makefile:116: package/feeds/routing/batctl/compile] Error 1
make[2]: Leaving directory '/home/chy/openwrt'
make[1]: *** [package/Makefile:110: /home/chy/openwrt/staging_dir/target-aarch64_cortex-a53_musl/stamp/.package_compile] Error 2
make[1]: Leaving directory '/home/chy/openwrt'
make: *** [/home/chy/openwrt/include/toplevel.mk:230: world] Error 2

**Comment by Boos4721** [comment, marked as answer]
**FIXED**
